powertools-common/src/test/java/software/amazon/lambda/powertools/common/internal/ClassPreLoaderTest.java

```diff
@@ -0,0 +1,34 @@
+package software.amazon.lambda.powertools.common.internal;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+class ClassPreLoaderTest {
+
+    // Making this volatile so the Thread Context doesn't need any special handling
+    static volatile boolean dummyClassLoaded = false;
```

Comment:
Out of interest: Can you explain what happens if don't put volatile here? I do not 100% follow the comment.

Reply:
I will not be able to modify the dummyClassLoaded as if its a local variable. I am setting it to false and then checking ClassPreLoader::preLoadClasses turns it to true if it loaded successfully. 
I did not witness in handful runs but theoretically there is a possibility of flakiness in tests as the variable change from classloader may not available immediately. 
Alternate option is to set and reset `Thread.currentThread().setContextClassLoader`.

At least thats my understanding, I ran the tests almost 7 times and it worked every time with or without volatile.

Reply:
I think I understand now:
1. Test initializes -> `dummyClassLoaded = false` is declared and initialized as `false`
2. ClassPreLoader runs
3. ClassPreLoader loads `DummyClass`
4. Static block triggers and sets `dummyClassLoaded = true` on the `ClassPreLoaderTest` class

It needs to be volatile such that the junit thread can immediatly assert the updated value based on the classloading of `DummyClass` executing the static block. 

Let me know if this makes sense.

Reply:
You've rightly put it!